specification/dimensions/commitmentdiscountid.md

```diff
@@ -0,0 +1,32 @@
+# Commitment Discount ID
+
+A commitment-based discount is a contractual commitment for an amount of usage or spend, for a specified term, in exchange for a discounted unit price. Commitment-based discounts can cover either usage or cost of various resources such as virtual machines or databases. A provider can offer various pricing models, payment options, and term durations for commitment-based discounts.
+
+A Commitment Discount ID is the identifier assigned to a commitment-based discount by the provider. Commitment Discount ID is commonly used for scenarios like chargeback for commitments and savings per commitment-based discount.
+
+The CommitmentDiscountId column MUST be present in the billing data. This column must be of type String and MUST NOT contain null values when a charge is related to a commitment-based discount. When a charge is not associated with a commitment-based discount, the column MUST be null. CommitmentDiscountId MUST be unique within the provider.
```

Comment:
Maybe it can be mentioned that CommitmentDiscountID works in pair with CommitmentDiscountName.

Reply:
While that is logically true for human-reading, an ID still retains its value as a dimension independent of the name. Would also look into how other ID/name dimensions are presented with this change, and I think lack of reference to each other makes sense.

Reply:
This is in keeping with our existing id/name pairs such as BillingAccountId, ResourceId, and SubAccountId. We might need to update those items as well.

Reply:
Yes indeed, nothing is mentioned for the id fields in FOCUS v0.5.
So this could be done later overall and in a consistent way.

Reply:
Yes, this was discussed earlier on - and we are going to move metrics and dimensions under a single 'columns' section and then add a section where we link 'related' columns and attributes. This will be done at the end - as a final pass. Good call-out though.